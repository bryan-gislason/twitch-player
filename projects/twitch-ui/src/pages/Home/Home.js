import { useTranslation } from "@twitch-player/i18n";
import PropTypes from "prop-types";
import React, { memo } from "react";
import { ScrollView } from "react-native";
import FeaturedChannels from "../../components/FeaturedChannels";
import TopCategories from "../../components/TopCategories";
import TopChannels from "../../components/TopChannels";

const Home = ({ testID = "home-page" }) => {
  const { t } = useTranslation();

  return (
    <ScrollView testID={testID}>
      <FeaturedChannels heading={t("pages.home.featuredChannelsHeading")} />
      <TopCategories heading={t("pages.home.topCategoriesHeading")} />
      <TopChannels heading={t("pages.home.topChannelsHeading")} />
      <TopChannels
        platformType="ps4"
        heading={t("pages.home.topPS4ChannelsHeading")}
      />
      <TopChannels
        platformType="xbox"
        heading={t("pages.home.topXboxOneChannelsHeading")}
      />
    </ScrollView>
  );
};

Home.propTypes = {
  testID: PropTypes.string,
};

export default memo(Home);
