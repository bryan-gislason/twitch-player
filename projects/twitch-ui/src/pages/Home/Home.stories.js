import { storiesOf } from "@storybook/react";
import { createClient } from "@twitch-player/data";
import React from "react";
import { ApolloProvider } from "react-apollo-hooks";
import Main from "../../components/Main";
import Home from ".";

storiesOf("Pages|Home", module).add("default", () => (
  <ApolloProvider client={createClient()}>
    <Main>
      <Home />
    </Main>
  </ApolloProvider>
));
