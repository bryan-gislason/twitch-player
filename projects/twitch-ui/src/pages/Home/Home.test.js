import React from "react";
import { render } from "../../testing";
import Home from "./Home";

jest.mock("../../components/FeaturedChannels", () => "mock-featured-channels");
jest.mock("../../components/TopCategories", () => "mock-top-categories");
jest.mock("../../components/TopChannels", () => "mock-top-channels");

describe("Home", () => {
  let instance;

  beforeEach(() => {
    instance = render(<Home />);
  });

  it("renders the page", () => {
    expect(instance.getByTestId("home-page")).toMatchSnapshot();
  });
});
