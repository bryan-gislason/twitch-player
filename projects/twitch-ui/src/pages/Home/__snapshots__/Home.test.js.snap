// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders the page 1`] = `
<div
  class="rn-WebkitOverflowScrolling-150rngu rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-flexGrow-16y2uox rn-flexShrink-1wbh5a2 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-overflowX-11yh6sk rn-overflowY-1rnoaur rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-transform-emqnss rn-zIndex-1lgpqti"
  data-testid="home-page"
>
  <div
    class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  >
    <mock-featured-channels
      heading="Featured Channels"
    />
    <mock-top-categories
      heading="Top Categories"
    />
    <mock-top-channels
      heading="Top Live Channels"
    />
    <mock-top-channels
      heading="Top PS4 Channels"
      platformtype="ps4"
    />
    <mock-top-channels
      heading="Top Xbox One Channels"
      platformtype="xbox"
    />
  </div>
</div>
`;
